packages/app/pages/course/[cid]/insights.tsx

```diff
@@ -15,21 +15,35 @@ import InsightsDisplayOptions from "../../../components/Insights/components/Insi
 export default function Insights(): ReactElement {
   const router = useRouter();
   const { cid } = router.query;
-  const { data: profile, error, mutate } = useSWR(`api/v1/profile`, async () =>
-    API.profile.index()
+  const { data: profile, mutate: mutateProfile } = useSWR(
+    `api/v1/profile`,
+    async () => API.profile.index()
+  );
+  const { data: allInsights } = useSWR(`api/v1/insights/listAll`, async () =>
+    API.insights.list()
   );
   const [settingsVisible, setSettingsVisible] = useState(false);
 
   const toggleInsightOn = async (insightName) => {
     await API.insights.toggleOn(insightName);
-    mutate();
+    mutateProfile();
   };
 
   const toggleInsightOff = async (insightName) => {
     await API.insights.toggleOff(insightName);
-    mutate();
+    mutateProfile();
   };
 
+  if (!allInsights || !profile?.insights) {
+    return null;
+  }
+  // Group users insights by size (small | default) so they can be rendered correctly
+  const [smallInsights, defaultInsights] = profile.insights.reduce(
+    ([s, d], i) =>
```

Comment:
Just for self actualization, What do s, d, and I represent?

Reply:
s -> smallInsights
d -> defaultInsights
i -> insight

Reply:
I just renamed for clarity. https://github.com/sandboxnu/office-hours/pull/522/commits/bb5f7e1394c9911f497359387e661de02dc1ed86